### PR TITLE
Add Grafana Faro

### PR DESCRIPTION
**Pull request by mdcruz**
On the Faro Grafana UI, please follow the instructions listed [here](https://grafana.com/docs/grafana-cloud/monitor-applications/frontend-observability/#create-an-application-in-grafana-cloud-frontend-observability) to set it up on your own Grafana instance.

You can find the URL on the Web SDK configuration view:

Screenshot 2023-08-04 at 18 21 11

Then, before running the docker container, export the `QUICKPIZZA_CONF_FARO_URL` env var to your container.

Here's an example session:

https://github.com/grafana/quickpizza/assets/10811379/927194b2-c207-4d13-9db4-598a9777586e

### CODE REVIEW

**Comment by roobre** [comment]
@mdcruz @dgzlopes I've pushed https://github.com/grafana/quickpizza/pull/21/commits/141d79f60b14a4ee41f8c1abbcc273f5ef45545b, that makes a small change to the init code to fetch the faro_url. I totally have no idea of what I'm doing with svelte, so please feel free to point out any javascript war crimes I might have commited.

File: .env.local [hunk 1]
```diff
@@ -1,2 +1,5 @@
 PUBLIC_BACKEND_ENDPOINT=http://localhost:3333/
-PUBLIC_BACKEND_WS_ENDPOINT=ws://localhost:3333/
\ No newline at end of file
+PUBLIC_BACKEND_WS_ENDPOINT=ws://localhost:3333/
+
+# copy to pkg/web .env file
+PUBLIC_FARO_WEBSDK_URL=your faro web URL
```
Comment (mdcruz) [inline .env.local:5]: We can now delete this, it's not needed anymore.

File: README.md [hunk 2]
```diff
@@ -119,6 +119,15 @@ docker run --rm -i -p 3333:3333 -e QUICKPIZZA_OTLP_ENDPOINT=http://grafana-agent
 
 The `QUICKPIZZA_TRUST_CLIENT_TRACEID` environment variable instructs QuickPizza to trust client-provided Trace IDs, so all client-generated spans (if any) will show up in the trace.
 
+## Enable Grafana Faro
+
+To enable Grafana Faro and monitor frontend observability, simply set `QUICKPIZZA_CONF_FARO_URL` to your Faro web URL:
```
Comment (mdcruz) [inline README.md:124]:
```suggestion
To enable Grafana Faro and monitor frontend observability, stop the QuickPizza container if it's already running and simply set `QUICKPIZZA_CONF_FARO_URL` to your Faro web URL:
```

File: pkg/web/src/hooks.client.ts [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+    then(config => {
+        const url = config.faro_url;
+        if (url === undefined) {
+            console.warn("Grafana faro is not configured.")
+            return
+        }
+
+        console.log(`Initializing Grafana Faro to ${url}`)
+        initializeFaro({
+            url: url,
```
Comment (mdcruz) [inline pkg/web/src/hooks.client.ts:17]:
```suggestion
            url,
```

File: pkg/web/src/hooks.client.ts [hunk 4]
```diff
@@ -0,0 +1,40 @@
+import {getWebInstrumentations, initializeFaro} from '@grafana/faro-web-sdk';
+import {TracingInstrumentation} from '@grafana/faro-web-tracing';
+import {PUBLIC_BACKEND_ENDPOINT} from '$env/static/public'
+
+function setupFaro() {
+    const res = fetch(`${PUBLIC_BACKEND_ENDPOINT}api/config`).
```
Comment (mdcruz) [inline pkg/web/src/hooks.client.ts:6]: I don't think res is being used, so we can remove it.

```suggestion
    fetch(`${PUBLIC_BACKEND_ENDPOINT}api/config`).
```

**Review by mdcruz — COMMENTED** [review]
Just a few nitpicks, thank you so much @roobre! 💜

File: pkg/web/src/hooks.client.ts (hunk 4 above)
Comment (roobre) [reply pkg/web/src/hooks.client.ts:6]: Good catch there!

File: pkg/web/src/hooks.client.ts (hunk 3 above)
Comment (roobre) [reply pkg/web/src/hooks.client.ts:17]: I am once again blind to JS magicks 🧙🏻‍♀️

File: .env.local (hunk 1 above)
Comment (roobre) [reply .env.local]: Absolutely right, it's now gone!

File: README.md (hunk 2 above)
Comment (roobre) [reply README.md]: 👍🏻 
I've also removed the "simply", as I like to follow the https://justsimply.dev/ advice. We can bring it back if there are stronger editorial reasons to use it.

**Comment by mdcruz** [comment]
@roobre I can't approved it since I opened the PR. Feel fee to approve and merge it  😊

**Review by dgzlopes — APPROVED** [review]
LGTM :heart:

**Pull request merged**